### ISSUE TITLE
Blank lines in formatted text lost on open

### ISSUE THREAD

**Issue by MarcSabatella** [musescore/MuseScore#19189]
### Issue type

Other type of issue

### Bug description

In a text element containing blank lines between paragraphs/stanzas, the blank lines disappear on save/reload if the text has formatting applied to it.

### Steps to reproduce

1. New score
2. Add text frame
3. Add text consisting of three lines: "aaa", a blank line, and "bbb"
4. While still in edit mode, press Ctrl+A to select the full text
5. Use Properties to set a point size of 12 pt (or something other than the default)
6. Save
7. Reopen

Result: the blank line has disappeared.

Also note, at step 5, the display of font size in the Properties panel displays "-" as if the select text is currently at different sizes.  This isn't actually the case, and it's possible this erroneous report in Properties is somehow related.

### MuseScore Version

OS: Debian GNU/Linux 11 (bullseye), Arch.: x86_64, MuseScore version (64-bit): 4.2.0-232380304, revision: github-musescore-musescore-0cf3725

### Regression

Yes, this used to work in MuseScore 3.x and now is broken

### Operating system

Debian 11, Windows

### Additional context

This also fails if applying the formatting to the paragraphs/stanzas individually.  But it works as expected if instead of selecting the text and applying the formatting that way, you apply the formatting to the element as a whole (eg, while not in edit mode).

This was originally reported by a user (my brother!) loading a MuseScore 3 score that had been formatted this way into MuseScore 4.  I verified that the blank lines are still present when loading into MuseScore 3, but not when loaded into MuseScore 4.  And verified I could reproduce it from scratch both in 4.1.1 and in current master.

Here is the original MuseScore 3 file: 
[American Taxation zip.zip](https://github.com/musescore/MuseScore/files/12448542/American.Taxation.zip.zip)

**Comment by rgreen5** [comment]
Perhaps some connection with #18553?

**Comment by cbjeukendrup** [comment]
See https://github.com/musescore/MuseScore/pull/22475#issuecomment-2063919806, which may or may not load to a solution to this.

**Comment by cbjeukendrup** [comment]
I've been looking into solving this. The new `PEDANTIC_WHITESPACE` option mentioned in the comment above doesn't help anything, at least not for this issue.

The problem is that newlines (or whitespace in general?) between two tags don't appear as `Characters` in `XmlReader::readXml()`/`XmlReader::htmlToString`. And that is caused by the fact that TinyXML discards it while parsing the file.

I can see basically two solutions:
1. **Patching TinyXML:** patch it to add the possibility to retrieve the verbatim content of an element. This would be used to replace the implementation of the `XmlReader::readXml()` method. It would work by storing pointers into the original XML data to the start and end of the element contents, and then building a std::string_view with these pointers.
Advantage: no file format change; reads existing scores correctly without having to correct them manually. 
Disadvantage: _significantly_ patching 3rd party stuff; messing with pointers

2. **Write newlines as `<br />`:** 
advantage: no patchwork; less dependencies on how exactly XML parsers handle whitespace
disadvantages: file format change, where compatibility with existing files might not come for free; risky reading logic adjustment; `readXml` becomes a weird method, losing its generality

@mike-spa @igorkorsukov What are your thoughts about this?

**Comment by mike-spa** [comment]
@cbjeukendrup I am relieved that you came to the same conclusion as I did back then. There is a third option that is ugly but I think we should consider:

3. Document the behaviour and tell users how to work around it.

That's essentially what I did [here](https://github.com/musescore/MuseScore/issues/18553#issuecomment-2202439443). It's one of those cases where the use case is very rare, the solution _very_ complicated (whichever direction we choose to go for), and the workaround very simple. Otherwise, I think I'd prefer option 2.

**Comment by bkunda** [comment]
@mike-spa @cbjeukendrup how do you think we should proceed with this?

It strikes me that the proposed option 2 is risky, so we shouldn't be trying it now for 4.4. 

If documentation is the immediate solution, then I'd probably vote for bumping this to another release.

**Comment by cbjeukendrup** [comment]
Yes, maybe that's a good decision: document the behaviour for now as a known issue, and solve it in a later release.

One more thought: perhaps patching the XML library is not a _super_ bad thing to do (especially since it _has_ already been patched before). This is just _some_ XML library made by some guy. It's really something different from more "official" packages like Qt, KDDockWidgets, etc.. That's not to say that it's not a decent library (despite its limitations), but it's just _some_ library, so we can just as well customise it a little bit, as long as it's clear what's from us and what's original, so that we can easily update it when a new release is created.
So personally I'm a bit more inclined towards proposed solution 1.
